digPostingDateExtractor/date_manipulation.py

```diff
@@ -0,0 +1,568 @@
+        if len(tuples) > 0:
+            (d, m, y) = tuples[0]
+            return DataManipulation.make_iso(y, m, d, format)
+
+        tuples = re.findall(DataManipulation.myproviderguide_regex_day, str)
+        if (len(tuples) > 0):
+            (m, d, y) = tuples[0]
+            return DataManipulation.make_iso(y, m, d, format)
+
+        tuples = re.findall(DataManipulation.sipsap_regex_day, str)
+        if (len(tuples) > 0):
+            (m, d, y) = tuples[0]
+            return DataManipulation.make_iso(y, m, d, format)
+
+        last_ditch_parse = DataManipulation.iso8601date(str, format)
+        if last_ditch_parse:
+            if format=='time':
+                return last_ditch_parse
+            else:
+                reparse_datetime = datetime.datetime.strptime(last_ditch_parse, "%Y-%m-%dT%H:%M:%S")
+                # reparse_datetime = dateutil.parser.parse(last_ditch_parse)
+                return reparse_datetime.date().isoformat()
+
+        if default_time and default_time != '':
+            if default_time > datetime(2000,1,1):
+                if format == 'time':
+                    return default_time.isoformat()
+                else:
+                    return default_time.date().isoformat()
+            else:
+                return ''
+        else:
+            return ''
+
+    @staticmethod
+    def date_created(posttime, crawl_time, format='time'):
+        """
+        return DataManipulation.date_created(getValue("posttime_extraction"), getValue("timestamp"))
+        Construct iso time from posttime and crawl_time.
+        :param posttime: extracted string of posted data, in arbitrary format.
+        :type posttime: string
+        :param crawl_time: epoch of time when the page was crawled, known to be inaccurate.
+        :type crawl_time: string
+        :return: if posttime can be parsed, then return it, otherwise return the crawl_time
+        """
+        default_time = None
+        try:
+            default_time = datetime.strptime(crawl_time, "%Y-%m-%dT%H:%M:%S")
+        except:
+            pass
+        parsed_date = DataManipulation.posttime_date(posttime, None, format)
+        if parsed_date == '':
+            parsed_date = DataManipulation.extract_relative_date(posttime, default_time, format)
+        return parsed_date
+    
+    @staticmethod
+    def iso8601date(date, date_format=None):
+        """Convert a date to ISO8601 date format
+        input format: YYYY-MM-DD HH:MM:SS GMT (works less reliably for other TZs)
+        or            YYYY-MM-DD HH:MM:SS.0
+        or            YYYY-MM-DD
+        or            epoch (13 digit, indicating ms)
+        or            epoch (10 digit, indicating sec)
+        output format: iso8601"""
+        date = date.strip()
+
+        if date_format:
+            try:
+                return datetime.strptime(date, date_format).isoformat()
+            except Exception:
+                pass
+
+        try:
```

Comment:
I feel like there should be a better way to handle parsing dates than chaining a bunch of exceptions.  Not necessary to address this right now, but I'll open an issue after merging the PR.